### PR TITLE
feat: introduce proxy service client and announcements

### PR DESCRIPTION
**Pull request by bianucci**

### CODE REVIEW

File: iris-client-bff/src/main/java/iris/client_bff/config/ProxyServiceConfig.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+@Data
+@RequiredArgsConstructor
+@ConfigurationProperties("proxy-service")
+public class ProxyServiceConfig {
+	private String epsName;
+	private String proxyTarget;
```
Comment (bianucci) [inline iris-client-bff/src/main/java/iris/client_bff/config/ProxyServiceConfig.java:12]: renamte to proxyTargetSubdomain

File: iris-client-bff/src/main/java/iris/client_bff/proxy/EPSProxyServiceServiceClient.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	}
+
+	@Data
+	@Builder
+	@NoArgsConstructor
+	@AllArgsConstructor
+	public static class AnnouncementDto {
+		private String domain;
+		private Instant expiresAt;
+		// TODO anything else?
```
Comment (bianucci) [inline iris-client-bff/src/main/java/iris/client_bff/proxy/EPSProxyServiceServiceClient.java:62]: add target proxy

File: iris-client-bff/src/main/java/iris/client_bff/proxy/EPSProxyServiceServiceClient.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+		var announcementDto = AnnouncementDto.builder()
+				.domain(domain)
+				.expiresAt(oneWeekFromNow)
+				.build();
+
+		// TODO use correct method name
+		var methodName = config.getEpsName()
+				+ "."
+				+ "announ....";
```
Comment (bianucci) [inline iris-client-bff/src/main/java/iris/client_bff/proxy/EPSProxyServiceServiceClient.java:44]: rename

File: iris-client-bff/src/main/resources/application.properties [hunk 4]
```diff
@@ -24,3 +24,6 @@ logging.level.de.healthIMIS=INFO
 logging.level.com.googlecode.jsonrpc4j=TRACE
 
 eps-client.clientUrl=https://localhost:5556/jsonrpc
+
+proxy-service.eps-name="private-proxy-1"
+proxy-service.proxy-target="proxy.smart-meeting.de"
```
Comment (bianucci) [inline iris-client-bff/src/main/resources/application.properties:29]: rename to subdomain, or similar

File: iris-client-bff/src/main/java/iris/client_bff/config/ProxyServiceConfig.java (hunk 1 above)
Comment (lucky-lusa) [reply iris-client-bff/src/main/java/iris/client_bff/config/ProxyServiceConfig.java]: done

File: iris-client-bff/src/main/java/iris/client_bff/proxy/EPSProxyServiceServiceClient.java (hunk 2 above)
Comment (lucky-lusa) [reply iris-client-bff/src/main/java/iris/client_bff/proxy/EPSProxyServiceServiceClient.java]: done

File: iris-client-bff/src/main/java/iris/client_bff/proxy/EPSProxyServiceServiceClient.java (hunk 3 above)
Comment (lucky-lusa) [reply iris-client-bff/src/main/java/iris/client_bff/proxy/EPSProxyServiceServiceClient.java]: done

File: iris-client-bff/src/main/resources/application.properties (hunk 4 above)
Comment (lucky-lusa) [reply iris-client-bff/src/main/resources/application.properties:29]: done

**Review by lucky-lusa — APPROVED** [review]

**Pull request merged**